src/test/java/jp/kusumotolab/kgenprog/ga/mutation/selection/DefaultVariantSelectionTest.java

```diff
@@ -144,26 +143,69 @@ public void testOrderOfVariants() {
         .doesNotContainSequence(9L, 19L, 7L, 8L, 17L);
   }
 
+  /**
+   * BulidFailedの個体が選択されるかどうかをテストする.
+   */
+  @Test
+  public void testBuildFailed() {
+    final int variantSize = 12;
+    final DefaultVariantSelection variantSelection =
+        new DefaultVariantSelection(variantSize, random);
+    final List<Variant> current = new ArrayList<>();
+    final List<Variant> generated = new ArrayList<>();
+
+    // 2個に1個はbuildFailedの個体にする
+    setupLists(current, generated, 10,
+        e -> e % 2 == 0 ? EmptyTestResults.instance : new TestResults());
+
+    final List<Variant> selectedVariants = variantSelection.exec(current, generated);
+
+    // 12個選択するが，buildSuccessが10個しかないため10個しか返ってこない
+    assertThat(selectedVariants).hasSize(10)
+        .allMatch(Variant::isBuildSucceeded);
+  }
+
   /**
    * Variantを生成するメソッド.Fitnessのみを指定する.
    *
    * @param fitness Variantの持つFitness
    * @return variant 生成したVariant
    */
   private Variant createVariant(final Fitness fitness) {
-    final Variant variant = new Variant(0, 0, null, null, null, fitness, null, null);
-    return variant;
+    final TestResults testResults = new TestResults();
+    return createVariant(fitness, 0, testResults);
   }
 
   /**
-   * Variantを生成するメソッド.FitnessとIdを指定する.
+   * Variantを生成するメソッド.Fitness, Id, TestResultsを指定する.
    *
    * @param fitness Variantの持つFitness
    * @param id Variantに固有の値
+   * @param testResults Variantのテスト情報
    * @return variant 生成したVariant
    */
-  private Variant createVariant(final Fitness fitness, final int id) {
-    final Variant variant = new Variant(id, 0, null, null, null, fitness, null, null);
+  private Variant createVariant(final Fitness fitness, final int id,
+      final TestResults testResults) {
+    final Variant variant = new Variant(id, 0, null, null, testResults, fitness, null, null);
     return variant;
   }
+
+  /**
+   * テスト用のリストを設定するメソッド.
+   *
+   * @param current variantSelectionに渡すcurrentリスト
+   * @param generated variantSelectionに渡すgeneratedリスト
+   * @param num リストに追加する要素の数
+   * @param testResultsCreator testResultsの渡し方を指定するFuntionオブジェクト（intを受け取りTestResultsを返す）
+   */
+  private void setupLists(final List<Variant> current, final List<Variant> generated, final int num,
+      final Function<Integer, TestResults> testResultsCreator) {
+    for (int i = 0; i < num; i++) {
+      final double divider = num;
+      final double value = (1.0d * (i + (i % 2))) / divider;
```

Comment:
上と似てる　`divider`に代入しなくても良い
```suggestion
      final double value = (1.0d * (i + (i % 2))) / (double) num;
```